### PR TITLE
Add: message when contributors list ends

### PR DESCRIPTION
**Pull request by yashovardhanagrawal**
The show more button disappears and message is shown

Closes https://github.com/coala/landing-frontend/issues/183

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by yashovardhanagrawal** [comment]
@jayvdb fixed the indentation problem

File: resources/js/Directives/getinvolvedDirective.js [hunk 1]
```diff
@@ -31,6 +31,13 @@ app.directive('getinvolved', ['$http', function ($http) {
             $scope.loadMore = function () {
                 $scope.totalDisplayed += 20;
             };
+
+            $scope.nomoreItems = function () {
+                if($scope.totalDisplayed > 288 ) {
```
Comment (jayvdb) [inline resources/js/Directives/getinvolvedDirective.js]: What is `288` ?

**Comment by yashovardhanagrawal** [comment]
`288` is the number of people. 
For displaying the end of the list message, we need to set a limit. In the code, there is no mention of the max number of people and also the list of people has not been updated and is fixed at `288`.
That's why I set the limit to `288`

**Comment by yashovardhanagrawal** [comment]
If you people want any other condition for maximum, tell me I will update it

**Comment by jayvdb** [comment]
Do you think the number of people in our organisation is a static number?

**Comment by yashovardhanagrawal** [comment]
We already have a lot of members and their names are not updated there. This list itself is quite old

**Comment by yashovardhanagrawal** [comment]
There are already 700+ members, we have a list of only 288, either we make both dynamic then that's a different option like we can do something like whenever someone joins his name is automatically listed, but for that will have to take into account a number of GitHub APIs.

Then that will be bigger and more complex issue taking into account that we are using AngularJS.

File: resources/js/Directives/getinvolvedDirective.js (hunk 1 above)
Comment (blazeu) [reply resources/js/Directives/getinvolvedDirective.js]: @yashovardhanagrawal you could check the length of `self.contributors` instead

**Comment by prashant0598** [comment]
@yashovardhanagrawal one way is that u can add members that are in core development groups and omit those who are in pending removal group and list cannot be a static number.

**Comment by yashovardhanagrawal** [comment]
@prashant0598 Rather than using APIs or any specific number, I am just thinking to add a counter to count the number of cards displayed. I am studying the backend for this, let's see 😄

**Comment by yashovardhanagrawal** [comment]
@jayvdb @prashant0598 @blazeu 
I have added a counter this might serve the purpose

Also, I am thinking to add an issue related to the update of the contributors database if you people find it worth 😄

**Comment by yashovardhanagrawal** [comment]
For the explanation, the counter goes like everytime a card is loaded it means a contributor is shown, hence the counter increases; which reaches a maximum limit when there are no more cards to display.
The value of the counter is used as the upper limit in the function nomoreItems.

**Comment by blazeu** [comment]
There's an simpler way of doing this.
`self.contributors` is an array of contributors and it has fixed length based on the data fetched from the API, use it.

**Comment by yashovardhanagrawal** [comment]
@blazeu

I tried it first like using it like that first, (using `self.contributors.length`) but it was showing this error in the console 

`TypeError: Cannot read property 'length' of undefined
    at controller (getinvolvedDirective.js:33)
    at Object.invoke (angular.js:4839)
    at Q.instance (angular.js:10692)
    at p (angular.js:9569)
    at angular.js:9977
    at angular.js:16643
    at m.$eval (angular.js:17958)
    at m.$digest (angular.js:17772)
    at m.$apply (angular.js:18066)
    at l (angular.js:12207)`

I think it is not of that form that why I tried this way. If you can help me out here then I would do it accordingly 😃

**Comment by blazeu** [comment]
```js
if (self.contributors) {
    ...
```

**Comment by yashovardhanagrawal** [comment]
@blazeu now you might review the code

File: resources/js/Directives/getinvolvedDirective.js [hunk 2]
```diff
@@ -31,6 +31,21 @@ app.directive('getinvolved', ['$http', function ($http) {
             $scope.loadMore = function () {
                 $scope.totalDisplayed += 20;
             };
+
+            $scope.counter = function () {
+                if(self.contributors) {
+                    var keys = Object.keys(self.contributors);
```
Comment (blazeu) [inline resources/js/Directives/getinvolvedDirective.js]: `self.contributors` is already an array.

`self.contributors.length` is enough.

File: resources/js/Directives/getinvolvedDirective.js [hunk 3]
```diff
@@ -31,6 +31,21 @@ app.directive('getinvolved', ['$http', function ($http) {
...
+                    return keys.length;
+                }
+                return 0;
+            };
+
+            $scope.nomoreItems = function () {
+                if($scope.totalDisplayed > $scope.counter ) {
+                    return 1;
+                }
+                return 0;
```
Comment (blazeu) [inline resources/js/Directives/getinvolvedDirective.js]:
```js
return $scope.totalDisplayed > self.contributors.length
```

but check for `self.contributors` first.

Comment (yashovardhanagrawal) [reply resources/js/Directives/getinvolvedDirective.js]: the ng-hide & ng-show functions here depend on 0 & 1, 

the earlier 0 & 1 can be removed though (used for counter)

**Comment by yashovardhanagrawal** [comment]
@blazeu is it fine now?

actually, I was being stupid here for sure
sorry 😛

**Comment by yashovardhanagrawal** [comment]
@jayvdb @blazeu is the PR okay or any other changes are required?
kindly tell me 😃

File: partials/tabs/people.html [hunk 4]
```diff
@@ -29,8 +29,8 @@ <h4 class="no-margin light">{{ person.nick }}</h4>
       <div class="child" ng-repeat="(key, value) in person.metadata">
         <div class="prop">{{key}}</div>
         <div class="val">{{value}}</div>
-      </dib>
-    </dib>
+      </div>
```
Comment (blazeu) [inline partials/tabs/people.html]: Ideally this should be in another commit

File: resources/js/Directives/getinvolvedDirective.js [hunk 5]
```diff
@@ -31,6 +31,12 @@ app.directive('getinvolved', ['$http', function ($http) {
             $scope.loadMore = function () {
                 $scope.totalDisplayed += 20;
             };
+
+            $scope.nomoreItems = function () {
+                if(self.contributors){
```
Comment (blazeu) [inline resources/js/Directives/getinvolvedDirective.js]: Add space before the first paren and after the second paren so it follows the style of the other files.

**Comment by yashovardhanagrawal** [comment]
@blazeu You might view it now

I also improved the formatting in other sections of the directive file.

**Comment by yashovardhanagrawal** [comment]
@blazeu I have separated those commits and going to file an issue and open a PR for the typo error in people.html

**Comment by yashovardhanagrawal** [comment]
I have added the issue: https://github.com/coala/landing-frontend/issues/240

you might assign it to me

**Comment by newbazz** [comment]
Rather than assigning it to yourself mark it as newcomer and let other newcomer solve it.

**Comment by yashovardhanagrawal** [comment]
@newbazz very well said. I was thinking to add this as a GCI task, unfortunately, I'm not a GCI Mentor yet.

**Comment by newbazz** [comment]
Atleast mark it as newcomer so any newcomer whos finding an issue can assign it to him/herself.

**Comment by yashovardhanagrawal** [comment]
I guess that right remains with the maintainers, not a newcomer like me 😛

On 07-Nov-2017 6:57 PM, "newbazz" <notifications@github.com> wrote:

Atleast mark it as newcomer so any newcomer whos finding an issue can
assign it to him/herself.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/coala/landing-frontend/pull/239#issuecomment-342481640>,
or mute the thread

**Comment by newbazz** [comment]
@yashovardhanagrawal okk i am also unaware :P

**Comment by prashant0598** [comment]
@yashovardhanagrawal if u have made another issue we can close this pr :smiley:  Also if u want to work on that issue u can assign it to yourself through corobo :wink:

**Comment by prashant0598** [comment]
@yashovardhanagrawal if u have made another issue we can close this pr :smiley: Also if u want to work on that issue u can assign it to yourself through corobo :wink:

**Comment by yashovardhanagrawal** [comment]
@prashant0598 that issue if for a different task, this PR remains and I think the proposed work for this PR is also completed by me. 😜 
About the issue, it is yet unlabelled and unlabelled issue are not meant to be assigned to newcomers so corobo cannot assign it to me.
You might view the whole conversation here, if you want to know more about what is the progress 😄

File: resources/js/Directives/getinvolvedDirective.js [hunk 6]
```diff
@@ -8,16 +8,17 @@ app.directive('getinvolved', ['$http', function ($http) {
             self.contributors
 
             function excludeBot(contributors) {
-                return contributors.filter(function(member){
+                return contributors.filter(function(member) {
```
Comment (jayvdb) [inline resources/js/Directives/getinvolvedDirective.js]: please do not modify lines unrelated to your assigned issue.

Comment (yashovardhanagrawal) [reply resources/js/Directives/getinvolvedDirective.js]: okay..  you might check the PR now

**Comment by yashovardhanagrawal** [comment]
PR is ready, suggested changes have been done.

**Comment by prnvdixit** [comment]
Also, again - try not to make changes to your "master" branch, the consequences of doing so are well documented [here](http://api.coala.io/en/latest/Developers/Newcomers_Guide.html#step-3-creating-a-fork-and-testing-your-changes) :wink: 👍

**Comment by yashovardhanagrawal** [comment]
@prnvdixit I will keep that in mind from next time

**Comment by yashovardhanagrawal** [comment]
I have updated the changes to be done in the PR

**Comment by prnvdixit** [comment]
ack e65c5cb9a8c2de65282b0613c07bf2af3f604b3d

**Comment by yashovardhanagrawal** [comment]
@jayvdb is this PR fine by you?

**Comment by yashovardhanagrawal** [comment]
@jayvdb updated it

**Comment by prnvdixit** [comment]
Remove ```:``` from issue reference :wink:

**Comment by yashovardhanagrawal** [comment]
@prnvdixit I removed it earlier this time I forgot while updating 😛 

anyway I updated it 😄

**Comment by jayvdb** [comment]
ack https://github.com/coala/landing-frontend/pull/239/commits/f5be5e004735666b3877f8ba7aa3023680a53ee4

**Comment by jayvdb** [comment]
@rultor merge

**Comment by jayvdb** [comment]
Needs to be rebased

**Comment by yashovardhanagrawal** [comment]
I rebased it

**Comment by jayvdb** [comment]
reack https://github.com/coala/landing-frontend/pull/239/commits/d3b4a81b4ad85ffcedf56520e3ecf0fb30df97d5

**Comment by jayvdb** [comment]
@rultor merge

**Comment by jayvdb** [comment]
No, you didnt rebase.  Probably because you created this PR using branch `master`, and you need to fetch and rebase on top of `upstream/master` or `origin/master`.

**Comment by yashovardhanagrawal** [comment]
@jayvdb sorry I didn't fetch the upstream master, just rebased my commit as it is, I have done it now

**Comment by yashovardhanagrawal** [comment]
I think I have to rebase it again as the #248 is closed

**Comment by yashovardhanagrawal** [comment]
@jayvdb CI has passed, you may merge it now

**Comment by jayvdb** [comment]
reack https://github.com/coala/landing-frontend/pull/239/commits/cdf9296c4be50f98fa013539fff8ccf211511df2

**Comment by jayvdb** [comment]
@rultor merge
